File: examples/ffi/lib/foreign.c


void test(machine *vm) {
    printf("This is a C foreign function!\n");
}

void __oak_add(machine *vm) {
    int a = machine_pop(vm);
    int b = machine_pop(vm);
    printf("This should print %d => ", a + b);
    machine_push(vm, a + b);
}

File: examples/ffi/lib/foreign.go

func test(vm *machine) {
	fmt.Println("This is a Go foreign function!")
}

func __oak_add(vm *machine) {
	a := vm.pop()
	b := vm.pop()
	fmt.Printf("This should print %v => ", a+b)
	vm.push(float64(a + b))
}